### ISSUE TITLE
Can I turn off character encoding for the builder?

### ISSUE THREAD

**Issue by toddgeist** [odata2ts/odata2ts#324]
I think I may not understand something about how the builder is supposed to work.

Here is my query function

```
export const getCustomers = async () => {
  const customers = await service
    .Customers()
    .query((builder, qCustomers) => {
      return builder.select("CustomerName").top(3);
    });


  return customers.data.value;
};
```
I expect this to return the top 3 records and just the Single 'CustomerName" field. Instead I get all the records back and all the fields.

Am I missing something?

Thanks

Todd

**Comment by toddgeist** [comment]
Ok, I know what is causing this.

The url that the builder generates looks like this `?%24top=3&%24select=CustomerName%2CCustCode%2CPrimaryKey`

The oData service I am using can't use the character encodings.  This url works fine.

 `?$top=3&$select=CustomerName,CustCode,PrimaryKey`
 
 So is there someway I can tell the builder to NOT encode $ and ,
 
 Thanks
 
 Todd

**Comment by texttechne** [comment]
Hi @toddgeist,
the builder itself provides an option for it in its constructor: See [ODataQueryBuilderConfig](https://github.com/odata2ts/odata2ts/blob/main/packages/odata-query-builder/src/ODataQueryBuilderModel.ts#L49C3-L49C12). We make extensive use of it in tests. Unfortunately you are currently not able to influence that setting when using the generated services. So this becomes a feature request.

**Comment by texttechne** [comment]
### Workaround ###
As workaround you could initialize your own query builder along these lines:
```ts
import { createQueryBuilderV4  } from "@odata2ts/odata-query-builder";

const cService = service.Customers();
const qObject = cService.getQueryObject()

const url = createQueryBuilderV4(cService.getPath(), qObject, { unencoded: true })
  .select("CustomerName")
  .top(3)
  .toString();

// then use the http client directly
const response = await httpClient.get(url)
```

**Comment by toddgeist** [comment]
Thank you for your response. 

I can use the workaround until the feature lands.

All the best :-)

**Comment by eluce2** [comment]
I'm working on a similar setup to Todd, but having trouble getting the qObject from the workaround example

```typescript
const service = new MapAppService(httpClient, baseUrl)
const filesService = service.files()
const qObject = filesService.getQueryObject()
//                           ^^ does not exist
```

`MapAppService` is the main generated service, but these are the only available properties on the `filesService`

![CleanShot 2024-10-15 at 15 16 03@2x](https://github.com/user-attachments/assets/bf958d1a-c652-4496-83d2-efee957f62c3)

Am I missing something about how to instantiate the main service?

**Comment by texttechne** [comment]
Hi @eluce2,
the example above was a bit pseudo code. You get the Q-object from the generated Q-file, not from the service.

**Comment by texttechne** [comment]
Hi @eluce2 & @toddgeist,

this feature has been released with [@odata2ts/odata2ts@0.39.0](https://github.com/odata2ts/odata2ts/releases/tag/%40odata2ts%2Fodata2ts%400.39.0).

You configure this at runtime for the main service along these lines:
```ts
const mainService = new MainService(httpClient, baseUrl, { noUrlEncoding: true });
```
